src/redux/actions/signupAction.js

```diff
@@ -0,0 +1,18 @@
+import axios from 'axios';
+import actionTypes from './actionTypes';
+
+export const setCurrentUser = payload => ({
+  type: actionTypes.SET_CURRENT_USER,
+  payload,
+});
+export default userData => dispatch => axios.post(`${process.env.BACKEND_URL}/api/users`, userData)
+  .then((res) => {
+    const {
+      username, firstname, lastname, email, image,
+    } = res.data.user;
+    dispatch(setCurrentUser({
+      username, firstname, lastname, email, image,
+    }));
+    localStorage.setItem('token', res.data.user.token);
```

Comment:
Nice work sinmi, However, we already have a name for setting the token on local storage, I suggest you go with the agreed name for consistency.